packages/esm-patient-chart-app/src/visit/widget/visits-overview.component.tsx

```diff
@@ -0,0 +1,238 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import dayjs from 'dayjs';
+import styles from './visits-overview.scss';
+import Button from 'carbon-components-react/es/components/Button';
+import DataTableSkeleton from 'carbon-components-react/es/components/DataTableSkeleton';
+import DataTable, {
+  Table,
+  TableCell,
+  TableContainer,
+  TableBody,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableExpandRow,
+  TableExpandedRow,
+  TableExpandHeader,
+} from 'carbon-components-react/es/components/DataTable';
+import { useTranslation } from 'react-i18next';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { Visit, getVisitsForPatient, createErrorHandler, OpenmrsResource } from '@openmrs/esm-framework';
+import { fetchEncounterByUuid } from './encounter.resource';
+import { StructuredListBody } from 'carbon-components-react';
+
+const visitsToShowCount = 5;
+
+interface EncounterViewProps {
+  encounter: OpenmrsResource;
+}
+
+const RenderEncounter: React.FC<EncounterViewProps> = ({ encounter }) => {
+  const { t } = useTranslation();
+  const [encounterDetails, setEncounterDetails] = useState<OpenmrsResource | null>(null);
+  const { uuid, encounterType } = encounter;
+
+  const formatVisitTime = (inputDate: any) => {
```

Comment:
This function can be extracted out of the component. No need to re-allocate on every render.